src/shared-components/dropdown/index.tsx

```diff
@@ -32,7 +32,7 @@ type DropdownProps = {
   /**
    * The currently selected option. Can mount as `null`
    */
-  value?: string;
+  value?: string | number | null;
```

Comment:
I think we should try to match the allowed types for `<option>` and `<select>`: `value?: string | number | readonly string[] | undefined` (Feel free to skip `readonly string[]` since it doesn't fit our use case.)